org.OpenT2T.Sample.SuperPopular.Lamp/Vera Dimmable Light Bulb/js/thingTranslator.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+
+var https = require('https');
+var q = require('q');
+
+// logs device state
+function logDeviceState(device) {
+    if (typeof (device) !== 'undefined') {
+        console.log('  device.name          : ' + device.name);
+        console.log('  device.props         : ' + device.props);
+    } else {
+        console.log('device is undefined');
+    }
+};
+
+// Helper method to send power state commands to Vera
+function sendPowerStateCommandToDevice(loadTarget, deviceId, relayServer, relaySessionToken, pkDevice) {
+    var deferred = q.defer();
+
+    // set up the url which needs:
+    //   the serial number of the hub
+    //   deviceid of the light bulb
+    //   target brightness
+    var path = '/relay/relay/relay/device/' + pkDevice + '/port_3480/data_request?id=action&output_format=json&DeviceNum=' + deviceId + '&serviceId=urn:upnp-org:serviceId:Dimming1&action=SetLoadLevelTarget&newLoadlevelTarget=' + loadTarget;
+
+    var options = {
+        protocol: 'https:',
+        host: relayServer,
+        path: path,
+        headers: {
+            'MMSSession': relaySessionToken
+        },
+        method: 'GET'
+    };
+
+    var req = https.request(options);
+
+    req.on('error', (e) => {
+        console.log('problem with request:' + e.message);
+        deferred.reject(e);
+    });
+
+    req.on('response', (res) => {
+        res.setEncoding('utf8');
+
+        res.on('data', (chunk) => {
+        });
+
+        res.on('end', () => {
+            if (res.statusCode != 200)
+            {
+                console.log(res);
+                deferred.reject(res);
```

Comment:
I feel we need to come up with common error object that OpenT2T translators will throw. On error, we do deferred.reject(e). And In Nest  helper I do deferred.reject(new Error("Error Code:" + res.statusCode)), if statusCode!=200.